### PR TITLE
chore(website): Add docs for v2

### PR DESCRIPTION
**Pull request by yevgenypats**
Continuation of https://github.com/cloudquery/cloudquery/pull/1837

Main Changes:
- Documentation in the main docs should be plugin specific free
- All tutorials and plugin specific docs moved to plugin repo under `plugins/source/x/README.md` and `plugins/source/x/docs`
- Add Source Plugin Release Stages
- Release Blog

### CODE REVIEW

File: website/pages/docs/troubleshooting.md [hunk 1]
```diff
@@ -14,14 +14,12 @@ First things first - feel free to join our new [Discord](https://discord.gg/2mPf
...
+1. Check out previous issues at [https://github.com/cloudquery/cloudquery](https://github.com/cloudquery/cloudquery) and open a new one if no previous one has been opened or resolved.
 
 ## Debugging
 
 ### Verbose Logging
 
-Usually the first step that will be needed to debug/resolve an issue is to run `cloudquery` with `-v` to enable verbose logging.
\ No newline at end of file
+Usually the first step that will be needed to debug/resolve an issue is to run `cloudquery` with `-v` to enable verbose logging.
+
```
Comment (erezrokah) [inline website/pages/docs/troubleshooting.md:24]: Should this be `--log-level debug`?

**Review by erezrokah — CHANGES_REQUESTED** [review]
Did a first pass as there are quite a few changes.

I think we still use `fetch` quite a lot, and we should either use `fetch` or `sync`, but not both.
Also maybe still keep a note about telemetry

File: website/pages/docs/reference/cli/cloudquery_generate.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
+---
+title: "generate"
+---
+## cloudquery generate
```
[the remark this answers is not in the document]
Comment (hermanschaaf) [reply website/pages/docs/reference/cli/cloudquery_generate.md:4]: It's coming back soon 😄

File: website/components/mdx/_getting-started-download-section.mdx [hunk 3]
```diff
@@ -13,7 +11,7 @@ You can download the precompiled binary from [releases](https://github.com/cloud
 <Tab>
 
 ```bash
-curl -L https://versions.cloudquery.io/latest/v1/cloudquery_linux_x86_64 -o cloudquery
+curl -L https://github.com/cloudquery/cloudquery/releases/download/cli-v1.2.0-pre.1/cloudquery_Linux_x86_64 -o cloudquery
```
Comment (erezrokah) [inline website/components/mdx/_getting-started-download-section.mdx:14]: We can revert these changes as we have a latest link now via Vercel redirects

Comment (yevgenypats) [reply website/components/mdx/_getting-started-download-section.mdx]: done

File: website/pages/docs/advanced-topics/docker.md [hunk 4]
```diff
@@ -16,7 +16,7 @@ CloudQuery uses a YAML file as the primary means of configuration, you can execu
 docker run \
```
Comment (bbernays) [inline website/pages/docs/advanced-topics/docker.md:16]: In line 13, we need to remove reference to `cloudquery init`

File: website/pages/docs/faq.md [hunk 5]
```diff
@@ -8,21 +8,6 @@ title: FAQ
...
-- When using [AWS Organizations](/plugins/aws#aws-organizations), if an account is deleted or removed from the org then the database entries for that account will be untouched.
-- When using [GCP Folders or Organizations](/plugins/gcp#configuration), if a project is deleted or removed from the folder, the database entries for that account will be untouched.
-
-#### What if I want to remove resources from accounts/projects that I'm no longer fetching from?
-
-You can use the `cloudquery provider purge [provider]` to delete old resources. For example, you can remove
-resources that weren't updated in the last three days with `cloudquery provider purge [provider] --last-update=72h --dry-run=false`.
-You can find the full details [here](https://www.cloudquery.io/docs/cli/commands/cloudquery_provider_purge).
+There are two types of `write` modes in destination plugins: `overwrite` and `append`, if in append data will always be added and never deleted. If in `overwrite` data will be upserted based on primary keys and it will be up to the user to setup recurring task to delete stale data.
```
Comment (bbernays) [inline website/pages/docs/faq.md:13]: There are now 3 modes I believe

File: website/pages/docs/plugins/source-plugin-release-stages.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+[Official](source) source plugins go through multiple release stages: Alpha, Beta and GA.
+
+## Comparison Matrix
+
+See the following comparison matrix to understand differences and expectations.
+
+|                                | Alpha | Beta                | GA                  |
+| ------------------------------ | ----- | ------------------- | ------------------- |
+| Spec (Config) Breaking Changes | Yes   | No                  | No                  |
```
Comment (bbernays) [inline website/pages/docs/plugins/source-plugin-release-stages.md:11]: I believe under GA, it should be semantic versioning

File: website/pages/docs/quickstart.mdx [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  # registry: github
+
+  ## Optional. Default: cloudquery/name
+  # path: cloudquery/aws
+
+  ## Optional. Default: ["*"] - all tables. We recommend to specify specific tables that you need to sync as this
+  ## will reduce the amount of data synced and improve performance.
+  # tables: ["*"]
+
+  ## Required. all destinations you want to sync data to.
```
Comment (bbernays) [inline website/pages/docs/quickstart.mdx:68]: I think we should show the `skip_tables` and possibly even populate with a few resources that were highlighted here: https://github.com/cloudquery/cloudquery-issues/issues/447

**Review by bbernays — CHANGES_REQUESTED** [review]
The following pages also have to have the examples updated:
- `/docs/advanced-topics/rate-limiting`
- `docs/advanced-topics/proxy-configuration`

File: website/pages/docs/core-concepts/plugins.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
+# Plugins
+
+CloudQuery has a pluggable architecture and relies on two types of plugins:
+
+- **Source plugin** - Responsible for extracting and transforming configuration from cloud-providers, SaaS apps, and other APIs ([Available source plugins](../plugins/sources)).
+- **Destination plugin** - Responsible for writing the data from the source plugins to various destinations such as databases, message queues and storage (Full list available [here](../plugins/destinations)).
```
Comment (bbernays) [inline website/pages/docs/core-concepts/plugins.md:6]: Links are broken in the deployment

**Review by bbernays — CHANGES_REQUESTED** [review]
Needs to be updated: `docs/tutorials/assume-role`

File: website/pages/docs/developers/codegen.md [hunk 9]
```diff
@@ -0,0 +1,8 @@
+# Generating resources
+
+Adding resources to a provider can sometimes be a tedious task, some resources can have more than hundreds of fields and relations, and adding them all can
+take a long time. To remedy this issue, we provide codegen utilities as part of our [plugin-sdk](https://github.com/cloudquery/plugin-sdk). codegen allows to easily generate more of the boilerplate code for tables from Go code.
+
+## Examples
+
+The best example would be to checkout how this is done in our of our official plugins such as [gcp](https://github.com/cloudquery/cloudquery/tree/main/plugins/source/gcp/codegen)
```
Comment (bbernays) [inline website/pages/docs/developers/codegen.md:8]: I think we need to document the commands/steps

File: website/pages/docs/cq-vs-others/google-cloud-asset-inventory.md [hunk 10]
```diff
@@ -4,8 +4,8 @@ Google Cloud Asset Inventory is the native GCP asset inventory.
 
 **Key Differences:**
 
-- **Resource Types**: CloudQuery [supports](https://hub.cloudquery.io) more than 80 types of resources while GCP currently supports about 140 types of resources (tables). Being an open-source project you can easily add the missing resources without being blocked by a vendor.
-- **Vanilla PostgreSQL**: CloudQuery uses vanilla PostgreSQL which makes it play nicely with the whole SQL eco-system and gives you the ability to re-use other tools like Grafana/BI. GCP Asset Inventory is using a proprietary query language and thus doesn't give you the ability to re-use other tools easily.
+- **Resource Types**: CloudQuery [supports](https://www.cloudquery.io/plugins/gcp/tables) more than 80 types of resources while GCP currently supports about 140 types of resources (tables). Being an open-source project you can easily add the missing resources without being blocked by a vendor.
```
[the remark this answers is not in the document]
Comment (yevgenypats) [reply website/pages/docs/cq-vs-others/google-cloud-asset-inventory.md]: fixed

File: website/pages/docs/plugins/sources.md [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+|------------------------------|-----------------------------------|-------------------------------------|-------|
+| [AWS][AWS]                   | [Releases][AWS-Releases]          | [Changelog][AWS-Changelog]          | Alpha |
+| [Azure][Azure]               | [Releases][Azure-Releases]        | [Changelog][Azure-Changelog]        | Alpha |    
+| [GCP][GCP]                   | [Releases][GCP-Releases]          | [Changelog][GCP-Changelog]          | Alpha |
+| [DigitalOcean][DigitalOcean] | [Releases][DigitalOcean-Releases] | [Changelog][DigitalOcean-Changelog] | Alpha |
+| [GitHub][GitHub]             | [Releases][GitHub-Releases]       | [Changelog][GitHub-Changelog]       | Alpha |
+| [Heroku][Heroku]             | [Releases][Heroku-Releases]       | [Changelog][Heroku-Changelog]       | Alpha |
+| [k8s][k8s]                   | [Releases][k8s-Releases]          | [Changelog][k8s-Changelog]          | Alpha |
+| [Okta][Okta]                 | [Releases][Okta-Releases]         | [Changelog][Okta-Changelog]         | Alpha |
+| [Terraform][Terraform]       | [Releases][Terraform-Releases]    | [Changelog][Terraform-Changelog]    | Alpha |
```
Comment (bbernays) [inline website/pages/docs/plugins/sources.md:19]: When this gets merged/released should't these plugins still be in at alpha?

**Comment by yevgenypats** [comment]
@bbernays addressed most of the comments, others will be in follow-up PRs so users can start using the new versions and it will be easier to review.
